lib/base-compiler.ts

```diff
@@ -3083,6 +3083,14 @@ export class BaseCompiler implements ICompiler {
         if (result.inputFilename) {
             result.inputFilename = utils.maskRootdir(result.inputFilename);
         }
+
+        if (result.asm) {
```

Comment:
On our machine running compile explorer the source was in `/tmp/compile-exploer-[some random chars]/main.cpp` for example. Thus the main.cpp would never match in the tree view with this to highlight source code to match teh asm.

Reply:
I 100% understand the reasoning, but this is probably not the best way to solve the problem. I'll have a think about it

Reply:
I'm not 100% following what's happening here, but we can't have this in the mainline version of CE; it's a pretty core thing that ought to be handled elsewhere.

Reply:
If I take the cmake template it will create a `main.cpp` with a `CMakeLists.txt`. On our server it would compile that file into a `/tmp` folder. While on godbolt it would be in the `/app` folder. Then on our server when we would hover over the C++ code it would not high light the equivalent asm. To solve that I had to do this code change. I'm not sure why. If there is a proper way to fix that I am completely open :)